ClassTypes/PowerShell/HyperV/check-hyperv-vm/Test-HypervVmConfig.ps1

```diff
@@ -0,0 +1,568 @@
+    
+    Write-Verbose "$($localUserName) part of Hyper-V Administrators group? $($isHypervAdmin)"
+    
+    return $isHypervAdmin
+}
+
+<#
+.SYNOPSIS
+Returns true if user is in the local Administrator group.
+#>
+function Get-LocalAdminStatus{
+    param(
+        [Parameter(Mandatory = $true)][string] $LocalUserName
+    )
+
+    $isAdmin = $null -ne $(Get-WmiObject win32_groupuser |  Where-Object { $_.groupcomponent -like '*"Administrators"' } | Where-Object { $_.PartComponent -like $LocalUserName })
+    Write-Verbose "$($localUserName) part of Administrators group? $($isAdmin)"
+  
+    return $isAdmin
+}
+
+<#
+.SYNOPSIS
+Returns list of local user accounts added to the Host VM.
+#>
+function Get-AddedLocalUser{
+    $addedLocalUsers = @(Get-LocalUser | Where-Object { [int]$($_.SID -split '-' | Select-Object -Last 1) -ge '1000' })
+
+    return $addedLocalUsers
+}
+
+###################################################################################################
+#
+# Main execution block.
+#
+
+try {
+    Write-Host "Verifying OS"
+    if ([System.Environment]::OSVersion.Platform -ne [System.PlatformID]::Win32NT) { Write-Error "Script applies to Windows only." }
+
+    # --------------------- Checking USER settings ---------------------------------------------------
+    Write-Host "******************************"
+    Write-Host "* Checking User Permissions  *"
+    Write-Host "******************************"
+
+    # Try to find other users on the machines, if there is one, then ask if they should be added to the "Hyper-V Administrators" group
+    $addedLocalUsers = @(Get-AddedLocalUser)
+    if ($addedLocalUsers.Count -gt 0) {
+ 
+        foreach ($localUser in $addedLocalUsers) {
+            $localUserName = $localUser | Select-Object -ExpandProperty Name
+ 
+            if ($(Get-LocalAdminStatus -LocalUserName $localUserName) -or $(Get-HypervAdminStatus -LocalUserName $localUserName)) {
+                Write-Host "Verified user '$($localUserName)' has permissions to use Hyper-V."
+            }
+            else {
+                if ($PSCmdlet.ShouldContinue("User '$($localUserName)' can not use Hyper-V.  Add user to Hyper-V Administrators Group?", "Add user to Hyper-V Administrators Group?", [ref] $YesToAll, [ref] $NoToAll )) {          
+                    Add-LocalGroupMember -Group $(Get-HyperVAdminGroup) -Member $localUser
+                    Write-Host "$($localUserName) added to Hyper-V Administrators group."   
+                }  
+            }
+        }
+    }
+    Write-Host "Permission check completed.`n"
+
+    # --------------------- Checking HYPER-V VM settings ---------------------------------------------------
+   
+
+    if ($ConfigFilePath) {
+        $configs = Get-ConfigurationSettings -SettingsFilePath $ConfigFilePath
+    }
+    else {
+        $configs = New-Object ConfigurationInfo
+    }
+
+    $vms = @(Get-VM)
```

Comment:
It looks like Get-VM is part of the hyperv powershell module, should that be listed/validated as a requirement somewhere?